pal/samples/esp8266/iothub_ll_telemetry_sample/iothub_ll_telemetry_sample.ino

```diff
@@ -0,0 +1,258 @@
+
+/* Define several constants/global variables */
+static const char* connectionString = DEVICE_CONNECTION_STRING;
+#define MESSAGE_COUNT 5 // determines the number of times the device tries to send a message to the IoT Hub in the cloud.
+static bool g_continueRunning = true; // defines whether or not the device maintains its IoT Hub connection after sending (think receiving messages from the cloud)
+static size_t g_message_count_send_confirmations = 0;
+
+IOTHUB_MESSAGE_HANDLE message_handle;
+size_t messages_sent = 0;
+const char* telemetry_msg = "test_message";
+
+// Select the Protocol to use with the connection
+#ifdef SAMPLE_MQTT
+    IOTHUB_CLIENT_TRANSPORT_PROVIDER protocol = MQTT_Protocol;
+#endif // SAMPLE_MQTT
+#ifdef SAMPLE_HTTP
+   IOTHUB_CLIENT_TRANSPORT_PROVIDER protocol = HTTP_Protocol;
+#endif // SAMPLE_HTTP
+
+IOTHUB_DEVICE_CLIENT_LL_HANDLE device_ll_handle;
+
+static int callbackCounter;
+int receiveContext = 0;
+
+/* -- receive_message_callback --
+ * Callback method which executes upon receipt of a message originating from the IoT Hub in the cloud. 
+ * Note: Modifying the contents of this method allows one to command the device from the cloud. 
+ */
+static IOTHUBMESSAGE_DISPOSITION_RESULT receive_message_callback(IOTHUB_MESSAGE_HANDLE message, void* userContextCallback)
+{
+    int* counter = (int*)userContextCallback;
+    const char* buffer;
+    size_t size;
+    MAP_HANDLE mapProperties;
+    const char* messageId;
+
+    // Message properties
+    if ((messageId = IoTHubMessage_GetMessageId(message)) == NULL)
+    {
+        messageId = "<null>";
+    }
+
+    // Message content
+    if (IoTHubMessage_GetByteArray(message, (const unsigned char**)&buffer, &size) != IOTHUB_MESSAGE_OK)
+    {
+        LogInfo("unable to retrieve the message data\r\n");
+    }
+    else
+    {
+        LogInfo("Received Message [%d]\r\n Message ID: %s\r\n Data: <<<%.*s>>> & Size=%d\r\n", *counter, messageId, (int)size, buffer, (int)size);
+        // If we receive the work 'quit' then we stop running
+        if (size == (strlen("quit") * sizeof(char)) && memcmp(buffer, "quit", size) == 0)
+        {
+            g_continueRunning = false;
+        }
+    }
+
+    /* Some device specific action code goes here... */
+    (*counter)++;
+    return IOTHUBMESSAGE_ACCEPTED;
+}
+
+
+/* -- send_confirm_callback --
+ * Callback method which executes upon confirmation that a message originating from this device has been received by the IoT Hub in the cloud.
+ */
+static void send_confirm_callback(IOTHUB_CLIENT_CONFIRMATION_RESULT result, void* userContextCallback)
+{
+    (void)userContextCallback;
+    // When a message is sent this callback will get envoked
+    g_message_count_send_confirmations++;
+    LogInfo("Confirm Callback");
+    // LogInfo("Confirmation callback received for message %lu with result %s\r\n", (unsigned long)g_message_count_send_confirmations, MU_ENUM_TO_STRING(IOTHUB_CLIENT_CONFIRMATION_RESULT, result));
```

Comment:
commented out code should be removed unless critical. However here, why not print out the result?